### DISCUSSION TITLE
Supported devices not recognized

### DISCUSSION THREAD

**Discussion by rodrigorsdev** [Koenkk/zigbee2mqtt#5700] [Devices]
Hello,

I bought some sensors supported by zigbee2mqtt, which are listed at https://www.zigbee2mqtt.io/information/supported_devices.html, however adding them generates the unsupported log. How can I add them without opening a issue?

Thanks.

**Comment by Koenkk** [comment]
What device is it? Can you provide the data/database.db of it?

**Reply by rodrigorsdev** [reply]
Theres a lot, one of than is 

HEIMAN Combustible gas sensor
--
`{"id":6,"type":"Router","ieeeAddr":"0xbc33acfffe03f324","nwkAddr":53917,"manufId":4619,"manufName":"HEIMAN","powerSource":"Mains (single phase)","modelId":"GASSensor-EF-3.0","epList":[1,242],"endpoints":{"1":{"profId":260,"epId":1,"devId":1026,"inClusterList":[0,3,1280,2821],"outClusterList":[25],"clusters":{"genBasic":{"attributes":{"modelId":"GASSensor-EF-3.0","manufacturerName":"HEIMAN","powerSource":1,"zclVersion":3,"appVersion":18,"stackVersion":2,"hwVersion":32,"dateCode":"2020.8.20"}},"ssIasZone":{"attributes":{"iasCieAddr":"0x00124b0014d9a6e0","zoneState":1}}},"binds":[],"configuredReportings":[],"meta":{}},"242":{"profId":41440,"epId":242,"devId":97,"inClusterList":[],"outClusterList":[33],"clusters":{},"binds":[],"configuredReportings":[],"meta":{}}},"appVersion":18,"stackVersion":2,"hwVersion":32,"dateCode":"2020.8.20","zclVersion":3,"interviewCompleted":true,"meta":{},"lastSeen":1610387747087}`

**Reply by Koenkk** [reply]
`GASSensor-EF-3.0` is not supported indeed (note that there a multiple devices that look similar but internals might slighly differ). That means you have to follow https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html but probably the following works:

```js
{
    zigbeeModel: ['GASSensor-EF-3.0'],
    model: 'GASSensor-EF-3.00',
    vendor: 'HEIMAN',
    description: 'Combustible gas sensor',
    fromZigbee: [fz.ias_gas_alarm_2],
    toZigbee: [],
    exposes: [e.gas(), e.battery_low(), e.tamper()],
},
```

if gas: true/false doesn't work properly try:

```js
{
    zigbeeModel: ['GASSensor-EF-3.0'],
    model: 'GASSensor-EF-3.00',
    vendor: 'HEIMAN',
    description: 'Combustible gas sensor',
    fromZigbee: [fz.ias_gas_alarm_1],
    toZigbee: [],
    exposes: [e.gas(), e.battery_low(), e.tamper()],
},
```

See https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html on how to modify devices.js

**Reply by rodrigorsdev** [reply]
It worked


thanks

**Comment by Leo2442926161** [comment]
Hi Zigbee2Mqtt Team @Koenkk , I am Leo from Heiman,

Since we have more devices now, I would like to ask do we need to update the configuration.yaml or you can do it? 
And is it possible to configure some commands and attributes in private cluster?
Thanks.

**Reply by Koenkk** [reply]
Yes both are possible. To add support for new devices follow https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html

**Reply by Leo2442926161** [reply]
> Yes both are possible. To add support for new devices follow https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html

Thank you very much for your quick reply and sharing.
What can we do if you add our new devices?

**Reply by Koenkk** [reply]
> What can we do if you add our new devices?

I don't understand what you mean, then the devices will be supported after following that guide.
